Tensile/TensileCreateLibrary.py

```diff
@@ -518,53 +523,45 @@ def buildKernelSourceAndHeaderFiles(results, outputPath):
         kernelsToWrite.append((err, src, header, kernelName))
 
         # Create list of files
+        key = pathJoin(kernelName)  # Default kernel name
         if filename:
-            filesToWrite[os.path.join(os.path.normcase(outputPath), filename)].append(
-                (err, src, header, kernelName)
-            )
-        elif globalParameters["MergeFiles"]:
-            kernelSuffix = ""
-            if globalParameters["NumMergedFiles"] > 1:
-                kernelSuffix = validKernelCount % globalParameters["NumMergedFiles"]
-
-            filesToWrite[
-                os.path.join(os.path.normcase(outputPath), "Kernels" + kernelSuffix)
-            ].append((err, src, header, kernelName))
-        else:
-            filesToWrite[os.path.join(os.path.normcase(outputPath), kernelName)].append(
-                (err, src, header, kernelName)
-            )
+            key = pathJoin(filename)
+        elif mergeFiles:
+            kernelSuffix = str(validKernelCount % numMergedFiles) if numMergedFiles > 1 else ""
+            key = pathJoin(f"Kernels{kernelSuffix}")
+
+        filesToWrite[key].append((err, src, header, kernelName))
```

Comment:
We might consider making this block of code a function (as well as the block above and below). We could call this `makeFilesToWrite` the block above `makeKernelsToWrite`. Maybe we define them inline like the functions below thought that makes them difficult to test.

Reply:
Done, however, it's called `collectFilesToWrite`